x-pack/plugins/ml/public/application/components/data_grid/feature_importance/use_classification_path_data.tsx

```diff
@@ -0,0 +1,169 @@
+  }));
+
+  if (baseline && predictedValue !== undefined && Number.isFinite(predictedValue)) {
+    // get the adjusted importance needed for when # of fields included in c++ analysis != max allowed
+    // if num fields included = num features allowed exactly, adjustedImportance should be 0
+    const adjustedImportance =
+      predictedValue -
+      mappedFeatureImportance.reduce(
+        (accumulator, currentValue) => accumulator + currentValue.importance!,
+        0
+      ) -
+      baseline;
+
+    mappedFeatureImportance.push({
+      [FEATURE_NAME]: i18n.translate(
+        'xpack.ml.dataframe.analytics.decisionPathFeatureBaselineTitle',
+        {
+          defaultMessage: 'baseline',
+        }
+      ),
+      [FEATURE_IMPORTANCE]: baseline,
+      absImportance: -1,
+    });
+
+    // if the difference is small enough then no need to plot the residual feature importance
+    if (adjustedImportance > 1e-5) {
+      mappedFeatureImportance.push({
+        [FEATURE_NAME]: i18n.translate(
+          'xpack.ml.dataframe.analytics.decisionPathFeatureOtherTitle',
+          {
+            defaultMessage: 'other',
+          }
+        ),
+        [FEATURE_IMPORTANCE]: adjustedImportance,
+        absImportance: 0, // arbitrary importance so this will be of higher importance than baseline
+      });
+    }
+  }
+  const filteredFeatureImportance = mappedFeatureImportance.filter(
+    (f) => f !== undefined
+  ) as ExtendedFeatureImportance[];
+
+  return buildDecisionPathData(filteredFeatureImportance);
+};
+
+export const buildClassificationDecisionPathData = ({
+  featureImportance,
+  currentClass,
+}: {
+  featureImportance: FeatureImportance[];
+  currentClass: string | undefined;
+}): DecisionPathPlotData | undefined => {
+  if (currentClass === undefined) return [];
+  const mappedFeatureImportance: Array<
+    ExtendedFeatureImportance | undefined
+  > = featureImportance.map((feature) => {
+    const classFeatureImportance = Array.isArray(feature.classes)
+      ? feature.classes.find((c) => c.class_name === currentClass)
+      : feature;
+    if (classFeatureImportance && typeof classFeatureImportance[FEATURE_IMPORTANCE] === 'number') {
+      return {
+        [FEATURE_NAME]: feature[FEATURE_NAME],
+        [FEATURE_IMPORTANCE]: classFeatureImportance[FEATURE_IMPORTANCE],
+        absImportance: Math.abs(classFeatureImportance[FEATURE_IMPORTANCE] as number),
+      };
+    }
+    return undefined;
+  });
+  const filteredFeatureImportance = mappedFeatureImportance.filter(
+    (f) => f !== undefined
+  ) as ExtendedFeatureImportance[];
+
+  return buildDecisionPathData(filteredFeatureImportance);
+};
+
+export const findMaxMin = (
+  data: DecisionPathPlotData,
+  getter: Function
+): { max: number; min: number } => {
+  let min = Infinity;
+  let max = -Infinity;
+  data.forEach((d) => {
+    const value = getter(d);
+    if (value > max) max = value;
+    if (value < min) min = value;
+  });
+  return { max, min };
+};
```

Comment:
Could this maybe replace by `d3.extent()`?

Reply:
Updated in https://github.com/elastic/kibana/pull/73561/commits/ef815d11a2bf334d703366daff3519b38b848e11